src/OIGenerator.cpp

```diff
@@ -18,40 +18,77 @@
 
 #include <glog/logging.h>
 
+#include <boost/core/demangle.hpp>
 #include <fstream>
+#include <unordered_map>
 #include <variant>
 
 #include "DrgnUtils.h"
 #include "OIUtils.h"
 
 namespace ObjectIntrospection {
 
+std::unordered_map<std::string, std::string>
+OIGenerator::oilStrongToWeakSymbolsMap(drgnplusplus::program& prog) {
+  static constexpr std::string_view strongSymbolPrefix =
+      "int ObjectIntrospection::getObjectSize<";
+  static constexpr std::string_view weakSymbolPrefix =
+      "int ObjectIntrospection::getObjectSizeImpl<";
+
+  std::unordered_map<std::string, std::pair<std::string, std::string>>
+      templateArgsToSymbolsMap;
+
+  auto symbols = prog.find_symbols_by_name(nullptr);
+  for (drgn_symbol* sym : *symbols) {
+    auto symName = drgnplusplus::symbol::name(sym);
+    auto demangled = boost::core::demangle(symName);
+
+    if (demangled.starts_with(strongSymbolPrefix)) {
```

Comment:
This is unlikely for our case since we control the sites, but I wanted to flag the behavior here in the case where multiple mangled names map to one demangled name. I think the last one in the symbol table is selected - is that alright?

It might be safer to at least log a warning in that case so we know if something fishy is happening.